pkg/core/annotations.go

```diff
@@ -35,3 +37,39 @@ func (c *client) bumpNonceAnnotation(s *v1alpha1.Service) {
 	annotations[buildAnnotation] = strconv.Itoa(i + 1)
 	s.Spec.RunLatest.Configuration.RevisionTemplate.SetAnnotations(annotations)
 }
+
+func (c *client) bumpNonceAnnotationForBuild(s *v1alpha1.Service) {
+	configurationSpec := s.Spec.RunLatest.Configuration
+	build := getBuild(configurationSpec)
+	if build != nil {
+		annotations := build.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		nonce := annotations[buildAnnotation]
+		i, err := strconv.Atoi(nonce)
```

Comment:
rather than futz with conversation and errors, can we simply assign a unique value like `build.Generation` or `build.ResourceVersion` if you'd rather a string. Both values are read-only (set by the api server) and guaranteed unique between mutations.

Reply:
again, just keeping the same logic we already had, maybe we can create a separate PR if we want to change this